ui/app/pages/routes/index.js

```diff
@@ -206,6 +206,10 @@ class Routes extends Component {
       }
       : null
 
+    const sidebarShouldClose = sidebarTransaction &&
+      !sidebarTransaction.status === 'failed' &&
```

Comment:
I suspect that this would result in the sidebar staying open if it went from `pending` to `failed`, which seems... bad. :thinking:

Reply:
We could move `sidebarShouldClose` into component state (along with the status), then compare in `getDerivedStateFromProps` the current status to the previous status to determine whether such a transition occurred, or whether it was `failed` from the start.

Or even better, we could move it into Redux. We could respond to `UPDATE_METAMASK_STATE` in the `appState` store. If the sidebar is open, and if the sidebar transaction has just transitioned from a pending state, then we could set the sidebar to closed directly.

In general I think we should be responding to background state changes in Redux, not in the UI. Either in the reducers or the actions. In this case, responding in the reducer seems to make the most sense.

Reply:
Well... given that the impact of this regression is minor, I think it'd be fine to leave it if we think we can fix it within the week (before the RC is released). Then we can revert as a fallback, if that timeline doesn't work out.

Reply:
yeah, I don't think a revert is necessary. This will happen very very rarely, but should still be fixed.

Reply:
okay we can spend some time this week looking at this